### PR TITLE
Add identicon

### PR DESCRIPTION
**Pull request by Thykof**

### CODE REVIEW

File: src/components/Identicon/Identicon.tsx [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+export interface IdenticonProps extends ComponentPropsWithoutRef<'img'> {
+  username: string;
+  size?: number;
+  saturation?: number;
+  lightness?: number;
+  customClass?: string;
+}
+
+export const Identicon = (props: IdenticonProps) => {
```
Comment (maxime-carabina) [inline src/components/Identicon/Identicon.tsx:17]:
```suggestion
export function Identicon(props: IdenticonProps) {
```

**Review by pivilartisant — APPROVED** [review]

**Review by maxime-carabina — APPROVED** [review]
Change `const` to `function` and it's ready to be merged 🎉

File: src/components/Identicon/Identicon.tsx [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+import { ComponentPropsWithoutRef, useMemo } from 'react';
+
+import { identicon } from 'minidenticons';
+
+export interface IdenticonProps extends ComponentPropsWithoutRef<'img'> {
+  username: string;
+  size?: number;
+  saturation?: number;
+  lightness?: number;
```
Comment (mazmassa) [inline src/components/Identicon/Identicon.tsx:13]: Even that is optional, we'll use it!?

File: src/components/Identicon/Identicon.tsx [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+  return (
+    <img
+      data-testid="identicon"
+      className={`bg-neutral rounded-full ${customClass}`}
+      src={svgURI}
+      alt={username}
+      {...rest}
+      width={computedSize}
+      height={computedSize}
```
Comment (mazmassa) [inline src/components/Identicon/Identicon.tsx:37]:
```suggestion
      width={computedSize}
      height={computedSize}
      {...rest}
```

File: src/components/Identicon/Identicon.tsx [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+  const { username, size, saturation, lightness, customClass, ...rest } = props;
+
+  const computedSize = size?.toString() || '64';
+
+  const svgURI = useMemo(
+    () =>
+      'data:image/svg+xml;utf8,' +
+      encodeURIComponent(identicon(username, saturation, lightness)),
+    [username, saturation, lightness],
+  );
```
[the remark this answers is not in the document]
Comment (Thykof) [reply src/components/Identicon/Identicon.tsx:27]: "useMemo is a React Hook that lets you cache the result of a calculation between re-renders."

File: src/components/Identicon/Identicon.tsx (hunk 2 above)
Comment (Thykof) [reply src/components/Identicon/Identicon.tsx:13]: maybe? for now I have no idea, UI requirement is default value saturation 50% and lightness 50%

File: src/components/Identicon/Identicon.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (mazmassa) [reply src/components/Identicon/Identicon.tsx:32]: Yep! That I am aware. What I am having a hard time to understand is why do we need all that to render a autogenerated SVG!?

I am not against, it's just to understand.

File: src/components/Identicon/Identicon.tsx (hunk 2 above)
Comment (mazmassa) [reply src/components/Identicon/Identicon.tsx:13]: Oh really!? I am afraid we don't need it. Let you decide, then!

File: src/components/Identicon/Identicon.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (Thykof) [reply src/components/Identicon/Identicon.tsx:32]: we will use the icon profile almost in every page. I think it's a good idea to put that in a cache

**Review by mazmassa — APPROVED** [review]

**Pull request merged**